src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -980,7 +980,10 @@ private CodeTypeBase CreateModelDeclarations(OpenApiUrlTreeNode currentNode, Ope
     private CodeTypeBase CreateCollectionModelDeclaration(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, OpenApiOperation operation, CodeNamespace codeNamespace, string typeNameForInlineSchema, bool isRequestBody)
     {
         CodeTypeBase type = GetPrimitiveType(schema?.Items, string.Empty);
-        if (string.IsNullOrEmpty(type?.Name))
+        bool isEnumCollectionType =    (schema?.Items.Enum.Any() ?? false) //the collection could be an enum type so override with strong type instead of string type.
+                                    || ((schema?.Items.IsAnyOf() ?? false) || (schema?.Items.IsOneOf() ?? false)) && string.IsNullOrEmpty(schema?.Items.Format);//the collection could be a composed type with an enum type so override with strong type instead of string type.
```

Comment:
I'm not sure I understand why we also check the format here?

Reply:
I pulled this one from line 960 from the `CreateModelDeclarations` method. Essentially trying to stay consistent as the `CreateComposedModelDeclaration` is also guarded from the presence of a `format`